src/sagemaker/jumpstart/curated_hub/content_copy.py

```diff
@@ -0,0 +1,306 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""This module contains utilities related to SageMaker JumpStart."""
+from __future__ import absolute_import
+from typing import List, Set
+
+from concurrent import futures
+from dataclasses import dataclass
+from botocore.client import BaseClient
+from botocore.exceptions import ClientError
+
+from sagemaker.jumpstart.curated_hub.utils import (
+    find_objects_under_prefix,
+)
+from sagemaker.jumpstart.types import JumpStartModelSpecs
+from sagemaker.jumpstart.curated_hub.accessors.jumpstart_s3_accessor import JumpstartS3Accessor
+from sagemaker.jumpstart.curated_hub.accessors.s3_object_reference import (
+    S3ObjectReference,
+    create_s3_object_reference_from_bucket_and_key,
+)
+
+
+EXTRA_S3_COPY_ARGS = {"ACL": "bucket-owner-full-control", "Tagging": "SageMaker=true"}
+
+
+@dataclass
+class CopyContentConfig:
+    """Property class to assist S3 copy"""
+
+    src: S3ObjectReference
+    dst: S3ObjectReference
+    display_name: str
```

Comment:
nit: consider renaming `display_name` as it is not very intuitive w.r.t the class intent. Would `logging_name` be better?